sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -233,4 +257,136 @@ public void populateDisplayData(DisplayData.Builder builder) {
     }
   }
 
+  /**
+   * A {@link PTransform} to write into Apache Cassandra. See {@link CassandraIO} for details on
+   * usage and configuration.
+   */
+  @AutoValue
+  public abstract static class Write<T> extends PTransform<PCollection<T>, PDone> {
+
+    @Nullable abstract List<String> hosts();
+    @Nullable abstract Integer port();
+    @Nullable abstract String keyspace();
+    @Nullable abstract Class<T> entity();
+    @Nullable abstract CassandraService<T> cassandraService();
+    abstract Builder<T> builder();
+
+    public Write<T> withHosts(List<String> hosts) {
+      checkArgument(hosts != null, "CassandraIO.write().withHosts(hosts) called with null hosts");
+      checkArgument(hosts.isEmpty(), "CassandraIO.write().withHosts(hosts) called with empty "
+          + "hosts list");
+      return builder().setHosts(hosts).build();
+    }
+
+    public Write<T> withPort(int port) {
+      checkArgument(port > 0, "CassandraIO.write().withPort(port) called with invalid port "
+          + "number (%d)", port);
+      return builder().setPort(port).build();
+    }
+
+    public Write<T> withKeyspace(String keyspace) {
+      checkArgument(keyspace != null, "CassandraIO.write().withKeyspace(keyspace) called with "
+          + "null keyspace");
+      return builder().setKeyspace(keyspace).build();
+    }
+
+    public Write<T> withEntity(Class<T> entity) {
+      checkArgument(entity != null, "CassandraIO.write().withEntity(entity) called with null "
+          + "entity");
+      return builder().setEntity(entity).build();
+    }
+
+    public Write<T> withCassandraService(CassandraService<T> cassandraService) {
+      checkArgument(cassandraService != null, "CassandraIO.write().withCassandraService"
+          + "(service) called with null service");
+      return builder().setCassandraService(cassandraService).build();
+    }
+
+    @Override
+    public void validate(PCollection<T> input) {
+      checkState(hosts() != null || cassandraService() != null,
+          "CassandraIO.write() requires a list of hosts to be set via withHosts(hosts) or a "
+              + "Cassandra service to be set via withCassandraService(service)");
+      checkState(port() != null || cassandraService() != null, "CassandraIO.write() requires a "
+          + "valid port number to be set via withPort(port) or a Cassandra service to be set via "
+          + "withCassandraService(service)");
+      checkState(keyspace() != null, "CassandraIO.write() requires a keyspace to be set via "
+          + "withKeyspace(keyspace)");
+      checkState(entity() != null, "CassandraIO.write() requires an entity to be set via "
+          + "withEntity(entity)");
+    }
+
+    @Override
+    public PDone expand(PCollection<T> input) {
+      input.apply(ParDo.of(new WriteFn<T>(this,
```

Comment:
Why do you need the SerializableFunction: you're already passing `this` to WriteFn, it can call getCassandraService on that